python/cudf/cudf/_lib/interop.pyx

```diff
@@ -99,31 +100,53 @@ cdef void dlmanaged_tensor_pycapsule_deleter(object pycap_obj):
     dlpack_tensor.deleter(dlpack_tensor)
 
 
-def to_arrow(Table input_table, object column_names, bool keep_index=True):
+cdef vector[column_metadata] gather_metadata(object metadata) except +:
```

Comment:
Should be `except *`. `except +` translates C++ exceptions to Python ones, whereas `except *` propagates them to the caller. Since we mark all external C++ functions `except +` in their `.pxd` definitions, we don't have to worry about doing that here.